### PR TITLE
Improve pass config between kyuubi and spark engine

### PR DESCRIPTION
**Pull request by ulysses-you**
ulysses-you [Closes #385](https://github.com/yaooqinn/kyuubi/pull/385)    Target Issue Test Plan [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

Make test more flexible and  readable.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @pan3793 @yaooqinn

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala [hunk 1]
```diff
@@ -41,10 +42,11 @@ trait WithSparkSQLEngine extends KyuubiFunSuite {
       s"jdbc:derby:;databaseName=$metastorePath;create=true")
     System.setProperty("spark.sql.warehouse.dir", warehousePath.toString)
     System.setProperty("spark.sql.hive.metastore.sharedPrefixes", "org.apache.hive.jdbc")
+    conf.foreach{case (k, v) => System.setProperty(k, v)}
+    SparkSQLEngine.kyuubiConf.loadFromSystemProperties()
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala:46]: Do we still need this? seems `KyuubiConf(loadSysDefault: Boolean = true)` is able to load sys props?

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala [hunk 2]
```diff
@@ -24,6 +24,7 @@ import org.apache.kyuubi.{KyuubiFunSuite, Utils}
 trait WithSparkSQLEngine extends KyuubiFunSuite {
   protected var spark: SparkSession = _
   protected var engine: SparkSQLEngine = _
+  protected def conf: Map[String, String] = Map.empty
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala:27]: Can we remove this default value? then we can detect improper impl at compile time

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala (hunk 1 above)
Comment (ulysses-you) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala:46]: We need this due to `SparkSQLEngine.kyuubiConf` is a global static field.

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala (hunk 2 above)
Comment (ulysses-you) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala]: Removed

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala (hunk 1 above)
Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala:46]:
```scala
conf.foreach{case (k, v) =>
  System.setProperty(k, v)
  SparkSQLEngine.kyuubiConf.set(k, v)
}
```

Is it much more clear?

Comment (ulysses-you) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/WithSparkSQLEngine.scala]: ok

**Comment by ulysses-you** [comment]
thanks merging to master for 1.1.0

**Pull request closed without merging**